openassessment/assessment/api/staff.py

```diff
@@ -56,6 +59,72 @@ def assessment_is_finished(submission_uuid, requirements):
     return True
 
 
+def on_start(submission_uuid):
+    """
+    Create a new staff workflow for a student item and submission.
+
+    Creates a unique staff workflow for a student item, associated with a
+    submission.
+
+    Args:
+        submission_uuid (str): The submission associated with this workflow.
+
+    Returns:
+        None
+
+    Raises:
+        StaffAssessmentInternalError: Raised when there is an internal error
+            creating the Workflow.
+
+    """
+    try:
+        submission = submissions_api.get_submission_and_student(submission_uuid)
+        workflow, __ = StaffWorkflow.objects.get_or_create(
+            course_id=submission['student_item']['course_id'],
+            item_id=submission['student_item']['item_id'],
+            submission_uuid=submission_uuid
+        )
+    except DatabaseError:
+        error_message = (
+            u"An internal error occurred while creating a new staff "
+            u"workflow for submission {}"
+            .format(submission_uuid)
+        )
+        logger.exception(error_message)
+        raise StaffAssessmentInternalError(error_message)
+
+
+def on_cancel(submission_uuid):
+    """
+    Cancel the staff workflow for submission.
+
+    Sets the cancelled_at field in staff workflow.
+
+    Args:
+        submission_uuid (str): The submission UUID associated with this workflow.
+
+    Returns:
+        None
+
+    """
+    try:
+        workflow = StaffWorkflow.objects.get(submission_uuid=submission_uuid)
+        workflow.cancelled_at = now()
+        workflow.save()
```

Comment:
If only one field is being updated, you can use `update_fields` for a slight performance increase: https://docs.djangoproject.com/en/1.8/ref/models/instances/#specifying-which-fields-to-save